### PR TITLE
Add downgrade CI

### PR DESCRIPTION
**Pull request by ErikQQY**
Downgrade CI

### CODE REVIEW

File: .github/workflows/Downgrade.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Downgrade
+on:
+  pull_request:
+    branches:
+      - master
```
Comment (ChrisRackauckas) [inline .github/workflows/Downgrade.yml:5]:
```suggestion
      - main
```

**Comment by ChrisRackauckas** [comment]
Get tests passing

**Comment by ChrisRackauckas** [comment]
Rebase onto master. @avik-pal just fixed the tests.

**Comment by ChrisRackauckas** [comment]
Looks like some bound is needed

File: Project.toml [hunk 2]
```diff
@@ -30,18 +30,18 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
...
-ConcreteStructs = "0.2"
-DiffEqBase = "6.126"
+ConcreteStructs = "0.2.2"
+DiffEqBase = "6.144"
 FastClosures = "0.3"
-FiniteDiff = "2"
-ForwardDiff = "0.10.3"
+FiniteDiff = "2.21"
+ForwardDiff = "0.10.36"
 LinearAlgebra = "1.9"
```
Comment (avik-pal) [inline Project.toml:40]:
```suggestion
LinearAlgebra = "1.10"
```

File: Project.toml [hunk 3]
```diff
@@ -30,18 +30,18 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
...
-Reexport = "1"
-SciMLBase = "2.7"
+FiniteDiff = "2.21"
+ForwardDiff = "0.10.36"
+LinearAlgebra = "1.10"
+MaybeInplace = "0.1.1"
+PrecompileTools = "1.2"
+Reexport = "1.2"
+SciMLBase = "2.23"
 StaticArrays = "1"
```
Comment (avik-pal) [inline Project.toml:45]:
```suggestion
StaticArrays = "1.8"
```

File: Project.toml [hunk 4]
```diff
@@ -30,18 +30,18 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
...
 [compat]
 ADTypes = "0.2.6"
-ArrayInterface = "7"
-ChainRulesCore = "1"
-ConcreteStructs = "0.2"
-DiffEqBase = "6.126"
+ArrayInterface = "7.7"
+ChainRulesCore = "1.18"
+ConcreteStructs = "0.2.2"
+DiffEqBase = "6.144"
```
Comment (avik-pal) [inline Project.toml:36]:
```suggestion
DiffEqBase = "6.146"
```

File: Project.toml [hunk 5]
```diff
@@ -30,18 +30,18 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
 
 [compat]
 ADTypes = "0.2.6"
-ArrayInterface = "7"
-ChainRulesCore = "1"
-ConcreteStructs = "0.2"
-DiffEqBase = "6.126"
+ArrayInterface = "7.7"
+ChainRulesCore = "1.18"
```
Comment (avik-pal) [inline Project.toml:34]:
```suggestion
ChainRulesCore = "1.21"
```

File: Project.toml [hunk 6]
```diff
@@ -30,18 +30,18 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
...
+FiniteDiff = "2.21"
+ForwardDiff = "0.10.36"
+LinearAlgebra = "1.10"
+MaybeInplace = "0.1.1"
+PrecompileTools = "1.2"
+Reexport = "1.2"
+SciMLBase = "2.23"
+StaticArrays = "1.8"
+StaticArraysCore = "1.4.2"
+julia = "1.10"
```
Comment (avik-pal) [inline Project.toml:47]:
```suggestion
FiniteDiff = "2.22"
ForwardDiff = "0.10.36"
LinearAlgebra = "1.10"
MaybeInplace = "0.1.1"
PrecompileTools = "1.2"
Reexport = "1.2"
SciMLBase = "2.23"
StaticArrays = "1.9"
StaticArraysCore = "1.4.2"
julia = "1.10"
```

**Comment by ChrisRackauckas** [comment]
Fails due to some allocations

**Comment by avik-pal** [comment]
Very confused about these failures. Not sure how any of the downstream packages is affecting allocations

File: Project.toml [hunk 7]
```diff
@@ -30,21 +30,21 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
 
 [compat]
 ADTypes = "0.2.6"
-ArrayInterface = "7"
-ChainRulesCore = "1"
-ConcreteStructs = "0.2"
-DiffEqBase = "6.126"
+ArrayInterface = "7.7"
+ChainRulesCore = "1.21"
+ConcreteStructs = "0.2.2"
```
Comment (ChrisRackauckas) [inline Project.toml:35]:
```suggestion
ConcreteStructs = "0.2.4"
```

File: Project.toml [hunk 8]
```diff
@@ -30,21 +30,21 @@ SimpleNonlinearSolveStaticArraysExt = "StaticArrays"
 
 [compat]
 ADTypes = "0.2.6"
-ArrayInterface = "7"
-ChainRulesCore = "1"
-ConcreteStructs = "0.2"
-DiffEqBase = "6.126"
+ArrayInterface = "7.7"
+ChainRulesCore = "1.21"
+ConcreteStructs = "0.2.4"
```
Comment (ChrisRackauckas) [inline Project.toml:35]:
```suggestion
ConcreteStructs = "0.2.3"
```

**Comment by ChrisRackauckas** [comment]
Fixed the downgrade. Can you look at the Bryoden failures? https://github.com/SciML/SimpleNonlinearSolve.jl/actions/runs/7856403406/job/21439489594?pr=110

**Comment by avik-pal** [comment]
Even the NonlinearSolve Broyden is failing exclusively on Mac. I will try sshing into one of the mac minis to see if I can reproduce this.

**Comment by avik-pal** [comment]
Can't reproduce the failure on the macminis

**Pull request merged**